app/src/main/java/com/example/photogallery/presentation/PhotoPageFragment.kt

```diff
@@ -0,0 +1,70 @@
+package com.example.photogallery.presentation
+
+import android.annotation.SuppressLint
+import android.net.Uri
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.WebChromeClient
+import android.webkit.WebView
+import androidx.appcompat.app.AppCompatActivity
+import com.example.photogallery.databinding.FragmentPhotoPageBinding
+
+class PhotoPageFragment : PollNotificationHandlerFragment() {
+    private lateinit var uri: Uri
+    private lateinit var binding: FragmentPhotoPageBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        uri = arguments?.getParcelable(ARG_URI) ?: Uri.EMPTY
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentPhotoPageBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    @SuppressLint("SetJavaScriptEnabled")
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.webView.settings.javaScriptEnabled = true
+        onSetProgressBar()
+        binding.webView.loadUrl(uri.toString())
+    }
+
+    private fun onSetProgressBar() {
```

Comment:
This is not a reaction to some event. So the "on" prefix doesn't fit. I would name it `setupProgressBar`